src/Controller/Api/FileController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Controller\Api;
+
+use App\Entity\File;
+use App\Message\DeleteFile;
+
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * CRUD API for File Entity.
+ */
+class FileController extends AbstractFOSRestController
+{
+    /**
+     * Delete a file.
+     *
+     * @param File                   $file           File entity instance.
+     * @param MessageBusInterface    $messageBus     Symfony messenger bus interface instance.
+     * @param EntityManagerInterface $entityManager  Entity manager interface instance.
+     *
+     * @Route("/api/file/{id}", name="pelagos_api_datasets_delete", methods={"DELETE"}, defaults={"_format"="json"})
+     *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     * @IsGranted("CAN_DELETE", subject="file")
```

Comment:
There is not voter who supports the file entity, so no one will vote on this, which is good for now.